### PR TITLE
RNA: Introduce `Button` component

### PR DESCRIPTION
**Pull request by renatoagds**
Implement `Button` based on Emerald guidelines from Figma (linked to the discussion here).

#### Changes proposed in this Pull Request:

* Implement `Button` component with all props related.
* Implement `Button` docs page.
* Add `--jp-red-70` color at `ThemeProvider`.

#### Jetpack product discussion

dRt1pX85T68o5UNYK6AZro-fi-88%3A0

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start Storybook
* Navigate to `Button` stories.
* Play with all stories, doc and variants.

#### Demo 💻

https://user-images.githubusercontent.com/1663717/161169117-32cf136e-252f-4ab2-bfb1-ef9ece5c7627.mp4

### CODE REVIEW

File: projects/js-packages/components/components/button/style.module.scss [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	border-radius: var(--jp-border-radius);
+
+	// Loading Mode
+	& > :global(.components-spinner) {
+		margin: 0 var(--spacing-base);
+	}
+
+	// Normal Size & Font
+	&.normal {
+		height: 40px;
```
Comment (retrofox) [inline projects/js-packages/components/components/button/style.module.scss:16]: Should we define these values based on `--spacing-base` too. p1648820233528369-slack-C02TQF5VAJD

Comment (renatoagds) [reply projects/js-packages/components/components/button/style.module.scss]: 7497f98

File: projects/js-packages/components/components/button/index.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+export const BUTTON_SIZES = {
+	NORMAL: 'normal',
+	SMALL: 'small',
+};
+
+export const BUTTON_VARIANTS = {
+	PRIMARY: 'primary',
+	SECONDARY: 'secondary',
+	LINK: 'link',
+	EXTERNAL_LINK: 'external-link',
```
Comment (retrofox) [inline projects/js-packages/components/components/button/index.jsx:20]: I'm not sure whether external-link should be considered a variation.

**Review by retrofox — APPROVED** [review]
LGTM

**Comment by leogermani** [comment]
r243094-wpcom

**Pull request merged**